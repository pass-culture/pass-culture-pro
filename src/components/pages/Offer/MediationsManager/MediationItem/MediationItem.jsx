import { Field, Form } from 'pass-culture-shared'
import React from 'react'

const MediationItem = ({ mediation }) => {
  const { id, isActive, thumbUrl } = mediation || {}
  return (
    <Form
      action={`/mediations/${id}`}
      handleSuccessNotification={null}
      isAutoSubmit
      name={`mediation-${id}`}
      patch={mediation}
      Tag="li">
      <img alt={`accroche-${id}`} disabled={!isActive} src={thumbUrl} />
      <br />
      <br />
      <div className="columns is-centered">
        <Field name="isActive" type="checkbox" />
      </div>
    </Form>
  )
}

export default MediationItem
