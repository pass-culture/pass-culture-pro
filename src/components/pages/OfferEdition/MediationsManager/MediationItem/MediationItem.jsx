import {Field, Form} from 'pass-culture-shared'
import React from 'react'
import PropTypes from "prop-types"

const MediationItem = ({ mediation }) => {
  const { id, isActive, thumbUrl } = mediation || {}
  return (
    <Form
      action={`/mediations/${id}`}
      handleSuccessNotification={null}
      isAutoSubmit
      name={`mediation-${id}`}
      patch={mediation}
      Tag="li"
    >
      <img
        alt={`accroche-${id}`}
        disabled={!isActive}
        src={thumbUrl}
      />
      <br />
      <br />
      <div className="columns is-centered">
        <Field
          name="isActive"
          type="checkbox"
        />
      </div>
    </Form>
  )
}

MediationItem.defaultProps = {
  mediation: { thumbUrl: '' }
}

MediationItem.propTypes = {
  mediation: PropTypes.shape(
    {
     id: PropTypes.string.isRequired,
     isActive: PropTypes.bool.isRequired,
     thumbUrl: PropTypes.string
    }
  ),
}

export default MediationItem
