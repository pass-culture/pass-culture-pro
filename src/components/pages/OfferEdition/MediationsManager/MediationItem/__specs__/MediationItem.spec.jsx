import React from 'react'
import { shallow } from 'enzyme'

import MediationItem from '../MediationItem'
import { Field, Form } from 'pass-culture-shared'

describe('components | OfferEdition | MediationItem', () => {
  let props

  beforeEach(() => {
    props = {
      mediation: {
        id: 'AE',
        isActive: true,
        thumbUrl: 'http://example.com/image.jpg',
      },
    }
  })

  it('should match the snapshot', () => {
    // when
    const wrapper = shallow(<MediationItem {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  describe('render', () => {
    it('should render a Form component with the right props', () => {
      // when
      const wrapper = shallow(<MediationItem {...props} />)

      // then
      const form = wrapper.find(Form)
      expect(form).toHaveLength(1)
      expect(form.prop('action')).toBe('/mediations/AE')
      expect(form.prop('handleSuccessNotification')).toBeNull()
      expect(form.prop('isAutoSubmit')).toBe(true)
      expect(form.prop('name')).toBe('mediation-AE')
      expect(form.prop('patch')).toStrictEqual({
        id: 'AE',
        isActive: true,
        thumbUrl: 'http://example.com/image.jpg',
      })
      expect(form.prop('Tag')).toBe('li')
    })

    it('should render an img element with right props', () => {
      // when
      const wrapper = shallow(<MediationItem {...props} />)

      // then
      const img = wrapper.find('img')
      expect(img).toHaveLength(1)
      expect(img.prop('alt')).toBe('accroche-AE')
      expect(img.prop('disabled')).toBe(false)
      expect(img.prop('src')).toBe('http://example.com/image.jpg')
    })

    it('should render a Field component with the right props', () => {
      // when
      const wrapper = shallow(<MediationItem {...props} />)

      // then
      const field = wrapper.find(Field)
      expect(field.prop('name')).toBe('isActive')
      expect(field.prop('type')).toBe('checkbox')
    })
  })
})
