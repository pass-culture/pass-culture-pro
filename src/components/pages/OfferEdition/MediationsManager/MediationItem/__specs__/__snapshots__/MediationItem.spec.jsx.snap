// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | OfferEdition | MediationItem should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MediationItem
    mediation={
      Object {
        "id": "AE",
        "isActive": true,
        "thumbUrl": "http://example.com/image.jpg",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "BlockComponent": null,
      "Tag": "li",
      "action": "/mediations/AE",
      "children": Array [
        <img
          alt="accroche-AE"
          disabled={false}
          src="http://example.com/image.jpg"
        />,
        <br />,
        <br />,
        <div
          className="columns is-centered"
        >
          <Field
            displayValue={[Function]}
            errors={null}
            layout="horizontal"
            name="isActive"
            size="normal"
            storeValue={[Function]}
            type="checkbox"
          />
        </div>,
      ],
      "className": null,
      "errorsPatch": Object {},
      "failNotification": "Formulaire non validé",
      "formPatch": Object {},
      "formatPatch": [Function],
      "handleFail": null,
      "handleFailNotification": null,
      "handleFailRedirect": null,
      "handleSuccess": null,
      "handleSuccessNotification": null,
      "handleSuccessRedirect": null,
      "isAutoSubmit": true,
      "name": "mediation-AE",
      "normalizer": null,
      "onEnterKey": null,
      "onEscapeKey": null,
      "onSubmit": null,
      "patch": Object {
        "id": "AE",
        "isActive": true,
        "thumbUrl": "http://example.com/image.jpg",
      },
      "successNotification": "Formulaire non validé",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "accroche-AE",
          "disabled": false,
          "src": "http://example.com/image.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Field
            displayValue={[Function]}
            errors={null}
            layout="horizontal"
            name="isActive"
            size="normal"
            storeValue={[Function]}
            type="checkbox"
          />,
          "className": "columns is-centered",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "displayValue": [Function],
            "errors": null,
            "layout": "horizontal",
            "name": "isActive",
            "size": "normal",
            "storeValue": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "BlockComponent": null,
        "Tag": "li",
        "action": "/mediations/AE",
        "children": Array [
          <img
            alt="accroche-AE"
            disabled={false}
            src="http://example.com/image.jpg"
          />,
          <br />,
          <br />,
          <div
            className="columns is-centered"
          >
            <Field
              displayValue={[Function]}
              errors={null}
              layout="horizontal"
              name="isActive"
              size="normal"
              storeValue={[Function]}
              type="checkbox"
            />
          </div>,
        ],
        "className": null,
        "errorsPatch": Object {},
        "failNotification": "Formulaire non validé",
        "formPatch": Object {},
        "formatPatch": [Function],
        "handleFail": null,
        "handleFailNotification": null,
        "handleFailRedirect": null,
        "handleSuccess": null,
        "handleSuccessNotification": null,
        "handleSuccessRedirect": null,
        "isAutoSubmit": true,
        "name": "mediation-AE",
        "normalizer": null,
        "onEnterKey": null,
        "onEscapeKey": null,
        "onSubmit": null,
        "patch": Object {
          "id": "AE",
          "isActive": true,
          "thumbUrl": "http://example.com/image.jpg",
        },
        "successNotification": "Formulaire non validé",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "accroche-AE",
            "disabled": false,
            "src": "http://example.com/image.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Field
              displayValue={[Function]}
              errors={null}
              layout="horizontal"
              name="isActive"
              size="normal"
              storeValue={[Function]}
              type="checkbox"
            />,
            "className": "columns is-centered",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "displayValue": [Function],
              "errors": null,
              "layout": "horizontal",
              "name": "isActive",
              "size": "normal",
              "storeValue": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
